GG 41675 Added vector similarity function (optional) & score functionality

manini-zs opened the pull request.

manini-zs committed:
Added the threshold parameter next to k to get results above a certain threshold

manini-zs committed:
[Partial] Added support for similarity function and returning score for vector queries

manini-zs committed:
Added support for a higher version of attrs to support langflow

Also removed a debug statement (print)

manini-zs committed:
Unused import removed that leads to ModuleNotFoundError

manini-zs committed:
Reverted the score handling

manini-zs committed:
Revert "Reverted the score handling"

This reverts commit 6d7299e12bd4fdd433b5f80e2001b720e0939418.

manini-zs committed:
Added support for defining Vector Similarity function in python thin client

manini-zs committed:
Update install.txt

manini-zs committed:
Merge branch 'master' into gg-41678-vector-similarity-python
Merging the master

manini-zs committed:
Moved changes from old branch to the new branch

manini-zs committed:
Fixed the misplacement of the bitmask check

manini-zs committed:
Matched the master

manini-zs committed:
Fixed the merge issue

isapego commented on pygridgain/__init__.py:
```diff
@@ -17,4 +17,4 @@
 from pygridgain.binary import GenericObjectMeta
 
-__version__ = '1.5.0-dev'
+__version__ = '1.6.0-dev'
```
The version should be updated in the separate commit, actually.

manini-zs committed:
Reverted the version and also added the score handling for aync function

manini-zs committed:
[Partial] Added support for backward compatibility with 8.9.17

manini-zs committed:
Fixed issues with backward compatibility

By default it uses the latest version of query index configuration. Only when protocol context is available or does not have the feature enabled do we use the old format

manini-zs commented:
This pull request is on the older branch. A new pull request has been created and approved in its place.
https://github.com/gridgain/python-thin-client/pull/77
